src/audio/kpb.c

```diff
@@ -384,9 +384,13 @@ static int kpb_prepare(struct comp_dev *dev)
 			break;
 		}
 		if (sink->sink->comp.type == SOF_COMP_SELECTOR) {
-			/* we found proper sink */
+			/* we found proper real time sink */
 			cd->rt_sink = sink;
-			break;
+		} else if (sink->sink->comp.type == SOF_COMP_HOST) {
+			/* we found proper host sink */
+			cd->cli_sink = sink;
+		} else {
+			continue;
```

Comment:
I don't think it's needed.

Reply:
@mrajwa why does KPB need to know where host is ? I dont understand why it has to understand the topology.

Reply:
@tlauda, why do you think it is not needed? How should we acquire sink address otherwise?
@lgirdwood, because we need to drain data somewhere right? So we need to get sink address. This is on KPB's sink list and as long as we work only with one client it is sufficient. With multiple of customers we need to introduce clients registration procedure.

Reply:
@mrajwa I mean continue label is not necessary.

Reply:
Ok, I agree the whole else condition is not necessary.